src/toggle-button/style.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { SYSTEM } from '../internal/environment';
+import customCssProps from '../internal/generated/custom-css-properties';
+import { ToggleButtonProps } from './interfaces';
+
+export function getToggleButtonStyles(style: ToggleButtonProps['style']) {
+  if (SYSTEM !== 'core') {
+    return {};
+  }
+
+  const properties = {
+    borderRadius: style?.root?.borderRadius,
+    borderWidth: style?.root?.borderWidth,
+    paddingBlock: style?.root?.paddingBlock,
+    paddingInline: style?.root?.paddingInline,
+    [customCssProps.styleBackgroundActive]: style?.root?.background?.active,
+    [customCssProps.styleBackgroundDefault]: style?.root?.background?.default,
+    [customCssProps.styleBackgroundDisabled]: style?.root?.background?.disabled,
+    [customCssProps.styleBackgroundHover]: style?.root?.background?.hover,
+    [customCssProps.styleBackgroundPressed]: style?.root?.background?.pressed,
+    [customCssProps.styleBorderColorActive]: style?.root?.borderColor?.active,
+    [customCssProps.styleBorderColorDefault]: style?.root?.borderColor?.default,
+    [customCssProps.styleBorderColorDisabled]: style?.root?.borderColor?.disabled,
+    [customCssProps.styleBorderColorHover]: style?.root?.borderColor?.hover,
+    [customCssProps.styleBorderColorPressed]: style?.root?.borderColor?.pressed,
+    [customCssProps.styleBoxShadowActive]: style?.root?.boxShadow?.active,
+    [customCssProps.styleBoxShadowDefault]: style?.root?.boxShadow?.default,
+    [customCssProps.styleBoxShadowDisabled]: style?.root?.boxShadow?.disabled,
+    [customCssProps.styleBoxShadowHover]: style?.root?.boxShadow?.hover,
+    [customCssProps.styleBoxShadowPressed]: style?.root?.boxShadow?.pressed,
+    [customCssProps.styleColorActive]: style?.root?.color?.active,
+    [customCssProps.styleColorDefault]: style?.root?.color?.default,
+    [customCssProps.styleColorDisabled]: style?.root?.color?.disabled,
+    [customCssProps.styleColorHover]: style?.root?.color?.hover,
+    [customCssProps.styleColorPressed]: style?.root?.color?.pressed,
+    [customCssProps.styleFocusRingBorderColor]: style?.root?.focusRing?.borderColor,
+    [customCssProps.styleFocusRingBorderRadius]: style?.root?.focusRing?.borderRadius,
+    [customCssProps.styleFocusRingBorderWidth]: style?.root?.focusRing?.borderWidth,
+  };
+
+  return Object.fromEntries(Object.entries(properties).filter(([, value]) => value !== undefined));
```

Comment:
Why do we need to filter out undefined props?

Reply:
I understand the diff, but disagree with the approach. Besides, there is a performance penalty when doing this: not only we need to iterate over all properties, we also need to construct a new object from that.

It is not very big impact, but it grows with the number of properties and components on the page. Imagine a future table with lots of nested styles, propagated to its internal checkboxes, etc. Then imagine that in every row there are a few inline buttons with styles. Currently, all these styles are not handled separately (e.g. in some style provider), but applied in the component directly. This can account for some ~10ms of processing time or even more on lower-performing devices for this transformation alone.